### ISSUE TITLE
Persistent Path to java.exe / javaws.exe

### ISSUE THREAD

**Issue by 1of16** [ojdkbuild/ojdkbuild#50]
While installing Oracles JRE a few new folder will be created, one of them is C:\Program Files\Common Files\Oracle\Java\javapath
javapath itself seems to be junction to C:\Program Files\Common Files\Oracle\Java\javapath_target_<id> where you can find a copy of java.exe, javaw.exe and javaws.exe.
To make is easier for people I often create shortcuts (e.g. via GPO) on the public-desktop like ":\Program Files\Common Files\Oracle\Java\javapath\javaws.exe http://example.com/example.jnlp"
This worked great so far.

While installing the ojdkbuild there is no similar path created, so I am a bit puzzled how to create persistent shortcuts.
I know I can use the path-entry, so "javaws.exe http://example.com/example.jnlp" could work, if there is no Oracles JRE installed at the same time (maybe you noticed, that oracle "rearranged" the path-variables, so Oracles JRE is on position 1, the ojdkbuild-path wouldn't work).
I also could use "%JAVA_HOME%\webstart\javaws.exe http://example.com/example.jnlp", but have to beg, that no one changed this variable.
Is there a better way to figure out the path to javaws.exe?
Or could you maybe create similar folders to java.exe and javaws.exe like Oracle did?

**Comment by ojdkbuild** [comment]
M, I am not familiar with this feature of Oracle installer. AFAIU what is required is some "stable" path to launchers that doesn't depend on the minor version of installed jdk. And it is better to have such path without the environment variables (or at least without the commonly used environment variables that can be overridden). At the same time I'd rather avoid dealing with junctions/symlinks in installer.

> Is there a better way to figure out the path to javaws.exe?

`%JAVA_HOME%\webstart\javaws.exe` is correct and also `javaws.exe` can optionally be added (appended) to `%PATH%`. But both variables can be easily overridden. Maybe to introduce some ojdkbuild-specific env var for that?

> Or could you maybe create similar folders to java.exe and javaws.exe like Oracle did?

I will look how this is done in Oracle installer to understand this better.

**Comment by 1of16** [comment]
> M, I am not familiar with this feature of Oracle installer. AFAIU what is required is some "stable" path to launchers that doesn't depend on the minor version of installed jdk. And it is better to have such path without the environment variables (or at least without the commonly used environment variables that can be overridden). At the same time I'd rather avoid dealing with junctions/symlinks in installer.
yes, you got it right :) and I can understand, that junctions/symlinks are not a perfect solution, but maybe not the worst either

> Maybe to introduce some ojdkbuild-specific env var for that?
sounds like a good plan, yes!

> I will look how this is done in Oracle installer to understand this better.
thank you!

**Comment by 1of16** [comment]
thank you, sounds great!
I hope it is not too much too ask, but would it be possible that you build a new 8u191 release? I would love to switch to your jdk this year and would need the new variable.

**Comment by ojdkbuild** [comment]
Hi, I don't want to do a new 8u191 release, instead I uploaded updated 8u191 MSI [here](https://drive.google.com/open?id=1GLP_FzB9rrDlrUzCFHOgmmviFQIdhR3N) (x86_64 version), please ping me if you need an x86 version. This version has some more minor changes ([HTTPS in webstart](https://github.com/ojdkbuild/ojdkbuild/issues/51) and some others) that are also going to be included into 8u201.

**Comment by 1of16** [comment]
Hi @ojdkbuild ,
awesome, thank you!
yes, a x86 version would be perfect.
Now I've got a solution for nearly all my java-based problems :)

**Comment by ojdkbuild** [comment]
No problem! Uploaded [x86 version](https://drive.google.com/open?id=1IeEzCgeYXwxIz_xOIClp1IunNJvhO5Xg). Going to keep these links alive at least until 8u201 is released.
